app/client-v2/src/stores/projectStore.ts

```diff
@@ -0,0 +1,132 @@
+import { getApiUrl } from "@/config";
+import {
+  COMPLETE_STATUS_TYPES,
+  type AnalysisStatus,
+  type ApiResponse,
+  type Project,
+  type ProjectSample,
+  type WorkerResponse,
+  WorkerResponseValueTypes
+} from "@/types/projectTypes";
+import { mande } from "mande";
+import { defineStore } from "pinia";
+import { Md5 } from "ts-md5";
+
+const baseApi = mande(getApiUrl(), { credentials: "include" });
+
+// TODO: add proper error handling. Maybe best to add error state attribute and watch accordingly cos of nested things interval/workers
+export const useProjectStore = defineStore("project", {
+  state: () => ({
+    basicInfo: {} as Pick<Project, "id" | "name" | "timestamp">,
+    fileSamples: [] as ProjectSample[],
+    projectHash: "",
+    isRun: false,
+    analysisStatus: {} as AnalysisStatus,
+    pollingIntervalId: null as ReturnType<typeof setInterval> | null
+  }),
+  getters: {
+    isReadyToRun: (state) =>
+      state.fileSamples.length > 0 && state.fileSamples.every((sample: ProjectSample) => sample.sketch && sample.amr),
+    isProjectComplete: (state) => {
+      const analysisStatusValues = Object.values(state.analysisStatus);
+      return (
+        analysisStatusValues.length > 0 && analysisStatusValues.every((value) => COMPLETE_STATUS_TYPES.includes(value))
+      );
+    },
+    numOfStatus: (state) => Object.keys(state.analysisStatus).length,
+    analysisProgressPercentage(state): number {
+      return Math.round(
+        (Object.values(state.analysisStatus).filter((value) => COMPLETE_STATUS_TYPES.includes(value)).length /
+          this.numOfStatus) *
+          100
+      );
+    }
+  },
+  actions: {
+    async getProject(id: string) {
+      this.$reset();
+      try {
+        const projectRes = await baseApi.get<ApiResponse<Project>>(`/project/${id}`);
+        this.basicInfo = {
+          id: projectRes.data.id,
+          name: projectRes.data.name,
+          timestamp: projectRes.data.timestamp
+        };
+        this.fileSamples = projectRes.data.samples;
+        this.projectHash = projectRes.data.hash || "";
+
+        if (projectRes.data.status) {
+          this.isRun = true;
+          this.analysisStatus = projectRes.data.status;
+        }
```

Comment:
I feel like this would be neater if it was constructing a Project object (as suggested above) and then setting it atomically in the state.

Reply:
good idea il do all this project store comments in the pt2 PR as it will have to restructure in that as well